server/src/main/java/org/elasticsearch/index/mapper/MapperUtils.java

```diff
@@ -24,17 +24,23 @@
 enum MapperUtils {
     ;
 
-    /** Split mapper and its descendants into object and field mappers. */
-    public static void collect(Mapper mapper, Collection<ObjectMapper> objectMappers, Collection<FieldMapper> fieldMappers) {
+    /**
+     * Splits the provided mapper and its descendants into object, field, and field alias mappers.
+     */
+    public static void collect(Mapper mapper, Collection<ObjectMapper> objectMappers,
+                               Collection<FieldMapper> fieldMappers,
+                               Collection<FieldAliasMapper> fieldAliasMappers) {
         if (mapper instanceof RootObjectMapper) {
             // root mapper isn't really an object mapper
         } else if (mapper instanceof ObjectMapper) {
             objectMappers.add((ObjectMapper)mapper);
         } else if (mapper instanceof FieldMapper) {
             fieldMappers.add((FieldMapper)mapper);
+        } else if (mapper instanceof FieldAliasMapper) {
+            fieldAliasMappers.add((FieldAliasMapper) mapper);
         }
```

Comment:
maybe we should add an else block that throws an exception like you added in other places?

Reply:
👍